ui-modules/blueprint-composer/app/components/catalog-selector/catalog-selector.directive.js

```diff
@@ -108,15 +144,21 @@ export function catalogSelectorSortFilter($filter) {
     }
 }
 
-function controller($scope, $element, $q, $uibModal, $log, $templateCache, paletteApi, paletteDragAndDropService, iconGenerator, composerOverrides) {
+function controller($scope, $element, $timeout, $q, $uibModal, $log, $templateCache, paletteApi, paletteDragAndDropService, iconGenerator, composerOverrides) {
+    this.$timeout = $timeout;
+
+    $scope.viewModes = PALETTE_VIEW_MODES;
+    $scope.viewOrders = ['name', 'type', 'id'];
+    
+    if (!$scope.state) $scope.state = {};
+    if (!$scope.state.viewMode) $scope.state.viewMode = PALETTE_VIEW_MODES.normal;
+    if (!$scope.state.currentOrder) $scope.state.currentOrder = 'name';
```

Comment:
Could you use curly brackets for the `if` statement please? That's to be consistent with the rest of the codebase and also make it easier to read